src/main/java/com/gotocompany/firehose/sink/grpc/GrpcSinkFactory.java

```diff
@@ -21,23 +31,64 @@
+                    .configure(SslContextBuilder.forClient().trustManager(certFile))
+                    .build();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     public static AbstractSink create(Map<String, String> configuration, StatsDReporter statsDReporter, StencilClient stencilClient) {
         GrpcSinkConfig grpcConfig = ConfigFactory.create(GrpcSinkConfig.class, configuration);
         FirehoseInstrumentation firehoseInstrumentation = new FirehoseInstrumentation(statsDReporter, GrpcSinkFactory.class);
         String grpcSinkConfig = String.format("\n\tService host: %s\n\tService port: %s\n\tMethod url: %s\n\tResponse proto schema: %s",
                 grpcConfig.getSinkGrpcServiceHost(), grpcConfig.getSinkGrpcServicePort(), grpcConfig.getSinkGrpcMethodUrl(), grpcConfig.getSinkGrpcResponseSchemaProtoClass());
         firehoseInstrumentation.logDebug(grpcSinkConfig);
-
-        ManagedChannel managedChannel = ManagedChannelBuilder.forAddress(grpcConfig.getSinkGrpcServiceHost(), grpcConfig.getSinkGrpcServicePort())
-                        .keepAliveTime(grpcConfig.getSinkGrpcArgKeepaliveTimeMS(), TimeUnit.MILLISECONDS)
-                        .keepAliveTimeout(grpcConfig.getSinkGrpcArgKeepaliveTimeoutMS(), TimeUnit.MILLISECONDS)
-                        .usePlaintext().build();
-
-        GrpcClient grpcClient = new GrpcClient(new FirehoseInstrumentation(statsDReporter, GrpcClient.class), grpcConfig, managedChannel, stencilClient);
-        firehoseInstrumentation.logInfo("GRPC connection established");
-
+        boolean isTlsEnable = grpcConfig.getSinkGrpcTlsEnable();
+        GrpcClient grpcClient;
+        if (isTlsEnable) {
+            String base64Cert = grpcConfig.getSinkGrpcRootCA();
+            SslContext sslContext = buildClientSslContext(base64Cert);
+            firehoseInstrumentation.logInfo("SSL Context created successfully.");
+            ManagedChannel managedChannel = NettyChannelBuilder.forAddress(grpcConfig.getSinkGrpcServiceHost(), grpcConfig.getSinkGrpcServicePort())
+                    .keepAliveTime(grpcConfig.getSinkGrpcArgKeepaliveTimeMS(), TimeUnit.MILLISECONDS)
+                    .keepAliveTimeout(grpcConfig.getSinkGrpcArgKeepaliveTimeoutMS(), TimeUnit.MILLISECONDS)
+                    .sslContext(sslContext)
+                    .build();
+            grpcClient = new GrpcClient(new FirehoseInstrumentation(statsDReporter, GrpcClient.class), grpcConfig, managedChannel, stencilClient);
+        } else {
+            ManagedChannel managedChannel = ManagedChannelBuilder.forAddress(grpcConfig.getSinkGrpcServiceHost(), grpcConfig.getSinkGrpcServicePort())
+                    .keepAliveTime(grpcConfig.getSinkGrpcArgKeepaliveTimeMS(), TimeUnit.MILLISECONDS)
+                    .keepAliveTimeout(grpcConfig.getSinkGrpcArgKeepaliveTimeoutMS(), TimeUnit.MILLISECONDS)
+                    .usePlaintext()
+                    .build();
```

Comment:
A lot of configuration seems to be common, can we extract out the common parts out of the if-else block.
eg 
```
   ManagedChannelBuilder managedChannelBuilder = NettyChannelBuilder
                    .forAddress(grpcConfig.getSinkGrpcServiceHost(), grpcConfig.getSinkGrpcServicePort())
                    .keepAliveTime(grpcConfig.getSinkGrpcArgKeepaliveTimeMS(), TimeUnit.MILLISECONDS)
                    .keepAliveTimeout(grpcConfig.getSinkGrpcArgKeepaliveTimeoutMS(), TimeUnit.MILLISECONDS);

      if (isTlsEnabled) {
          String base64Cert = grpcConfig.getSinkGrpcRootCA();
          SslContext sslContext = buildClientSslContext(base64Cert);
          firehoseInstrumentation.logInfo("SSL Context created successfully.");
          ManagedChannelBuilder.sslContext(sslContext);
      } else {
          ManagedChannelBuilder.usePlaintext();
      }
    
   grpcClient = new GrpcClient(new FirehoseInstrumentation(statsDReporter, GrpcClient.class), grpcConfig, managedChannelBuilder.build(), stencilClient);
```

Reply:
@sbchaos Initially, I had structured the code similarly by creating a common ManagedChannelBuilder instance, but encountered an error when attempting to reuse it across the if-else branches. Consequently, I opted to use separate variables within each block.